From f230e00aa6eecaf028028c757d5971c958da9ec7 Mon Sep 17 00:00:00 2001
From: Artem Borisov <dedsa2002@gmail.com>
Date: Tue, 6 Nov 2018 12:47:02 +0400
Subject: [PATCH 31/71] HACK: media: Restore pre-4.1 media_entity_desc

Our userspace camera blobs are incompatible with the new struct.

Change-Id: I0a3ca0a8ae99eca6b154d939187093fa8a45e0b5
Signed-off-by: Albert I <kras@raphielgang.org>
---
 include/uapi/linux/media.h | 41 +++++---------------------------------
 1 file changed, 5 insertions(+), 36 deletions(-)

diff --git a/include/uapi/linux/media.h b/include/uapi/linux/media.h
index 4890787731b8..d33c4d275df4 100644
--- a/include/uapi/linux/media.h
+++ b/include/uapi/linux/media.h
@@ -201,41 +201,6 @@ struct media_entity_desc {
 
 	union {
 		/* Node specifications */
-		struct {
-			__u32 major;
-			__u32 minor;
-		} dev;
-
-#if 1
-		/*
-		 * TODO: this shouldn't have been added without
-		 * actual drivers that use this. When the first real driver
-		 * appears that sets this information, special attention
-		 * should be given whether this information is 1) enough, and
-		 * 2) can deal with udev rules that rename devices. The struct
-		 * dev would not be sufficient for this since that does not
-		 * contain the subdevice information. In addition, struct dev
-		 * can only refer to a single device, and not to multiple (e.g.
-		 * pcm and mixer devices).
-		 *
-		 * So for now mark this as a to do.
-		 */
-		struct {
-			__u32 card;
-			__u32 device;
-			__u32 subdevice;
-		} alsa;
-#endif
-
-#if 1
-		/*
-		 * DEPRECATED: previous node specifications. Kept just to
-		 * avoid breaking compilation, but media_entity_desc.dev
-		 * should be used instead. In particular, alsa and dvb
-		 * fields below are wrong: for all devnodes, there should
-		 * be just major/minor inside the struct, as this is enough
-		 * to represent any devnode, no matter what type.
-		 */
 		struct {
 			__u32 major;
 			__u32 minor;
@@ -244,8 +209,12 @@ struct media_entity_desc {
 			__u32 major;
 			__u32 minor;
 		} fb;
+		struct {
+			__u32 card;
+			__u32 device;
+			__u32 subdevice;
+		} alsa;
 		int dvb;
-#endif
 
 		/* Sub-device specifications */
 		/* Nothing needed yet */
-- 
2.20.1

